lib/recommendations.js

```diff
@@ -0,0 +1,170 @@
+const assert = require('assert');
+const fs = require('fs');
+const uuid = require('uuid');
+const common = require('./common');
+const httph = require('./http_helper.js');
+const query = require('./query.js');
+
+// SQL Queries
+const select_recommendation_by_primary_key = query.bind(query, fs.readFileSync('./sql/select_recommendation_by_primary_key.sql').toString('utf8'), null);
+const insert_recommendation = query.bind(query, fs.readFileSync('./sql/insert_recommendation.sql').toString('utf8'), null);
+const update_recommendation = query.bind(query, fs.readFileSync('./sql/update_recommendation_details.sql').toString('utf8'), null);
+
+// Information about available resource types that recommendations can target
```

Comment:
Question for reviewer: 

We hardcode other things of this nature (e.g. webhook types) but should this be information in a database table so it's dynamic, with defaults provided in the `create.sql` migration script? Thoughts?

Reply:
Yeah if this is an interface, lets try and get away from hardcoded types, and handle it more dynamic. I'm fine if we define some sort of interface for a catagory that has certain properties, but I'd like to get away from things like how we tie "web" dynos to so many things. It shouldn't matter that it's named web for example, but maybe a catagory of web server vs worker catagories, if that makes sense.